tests/cli-misc.rs

```diff
@@ -110,6 +110,33 @@ fn upgrade_v2_metadata_to_v12() {
     });
 }
 
+// Verifies the conversion from separate files to a single settings.toml
+#[test]
+fn upgrade_toml_settings() {
+    setup(&|config| {
+        rustup_utils::raw::write_file(&config.rustupdir.join("version"), "2").unwrap();
+        rustup_utils::raw::write_file(&config.rustupdir.join("default"), "beta").unwrap();
+        rustup_utils::raw::write_file(&config.rustupdir.join("overrides"),
+                                      "a;nightly\nb;stable").unwrap();
+        rustup_utils::raw::write_file(&config.rustupdir.join("telemetry-on"), "").unwrap();
+        expect_err(config, &["rustup", "default", "nightly"],
+                   "rustup's metadata is out of date. run `rustup self upgrade-data`");
```

Comment:
This is testing an upgrade from metadata version `2`, ie. an old metadata version, for which we currently require an explicit upgrade (because it wipes the toolchains). I can add a separate test for *just* the toml upgrade, but it would be a subset of what is tested here.